clinica/iotools/converters/ixi_to_bids/ixi_to_bids_utils.py

```diff
@@ -0,0 +1,209 @@
+        .assign(
+            modality=lambda df: df.img_name_no_ext.apply(
+                lambda x: rename_ixi_modalities(x.split("-")[3])
+            )
+        )
+        .assign(field=lambda df: df.hospital.apply(lambda x: define_magnetic_field(x)))
+        .assign(session="ses-M000")
+    )
+    return df
+
+
+def select_subject_data(data_df: pd.DataFrame, subject: str) -> pd.DataFrame:
+    """Select subject images to copy based on IXI id"""
+    return data_df[data_df["subject"] == subject]
+
+
+def bids_filename_from_ixi(img: pd.Series) -> str:
+    return f"{img['bids_id']}_{img['session']}_{img['modality']}"
+
+
+def write_ixi_json_subject(writing_path: str, hospital: str, field: str) -> None:
+    with open(writing_path, "w") as f:
+        json.dump(
+            {
+                "InstitutionName": hospital,
+                "MagneticFieldStrength (T)": field,
+            },
+            f,
+            indent=4,
+        )
+
+
+def write_subject_no_dti(subject_df: pd.DataFrame, bids_path: Path) -> None:
+    """Copies all subject data but DTI"""
+    for _, row in subject_df.iterrows():
+        cprint(
+            f"Converting modality {row['modality']} for subject {row['subject']}.",
+            lvl="debug",
+        )
+        filename = bids_filename_from_ixi(row)
+        data_path = bids_path / row["bids_id"] / row["session"] / "anat"
+        data_path.mkdir(parents=True, exist_ok=True)
+        shutil.copy2(row["img_path"], f"{data_path}/{filename}.nii.gz")
+        write_ixi_json_subject(
+            f"{data_path}/{filename}.json", row["hospital"], row["field"]
+        )
+        # todo : 1 json per image or 1 json per subject ?
+
+
+def write_subject_dti_if_exists(
+    bids_path: Path, subject: str, data_directory: Path
+) -> None:
+    """Processes DTI data if found for a subject"""
+    if dti_paths := find_subject_dti_data(data_directory, subject):
+        cprint(f"Converting modality DTI for subject {subject}.", lvl="debug")
+        dti_to_save = merge_dti(dti_paths)
+        bids_id = bids_id_factory(StudyName.IXI).from_original_study_id(subject)
+        data_path = bids_path / bids_id / "ses-M000" / "dwi"
+        data_path.mkdir(parents=True, exist_ok=True)
+        filename = f"{bids_id}_ses-M000_dwi"
+        dti_to_save.to_filename(f"{data_path}/{filename}.nii.gz")
+        hospital = dti_paths[0].name.split("-")[1]
+        write_ixi_json_subject(
+            f"{data_path}/{filename}.json", hospital, define_magnetic_field(hospital)
+        )
+    else:
+        cprint(f"No DTI data was found for IXI subject {subject}.", lvl="warning")
+
+
+def find_subject_dti_data(data_directory: Path, subject: str) -> List[Path]:
+    pattern = subject + r"(-\w*){4}.nii.gz$"
+    return [
+        path
+        for path in data_directory.rglob(pattern="IXI*.nii.gz")
+        if re.search(pattern, str(path))
+    ]
+
+
+def merge_dti(dti_images: List[Path]):
```

Comment:
The return type is missing here.

Reply:
I was not sure what to put here since it is a nifti image. What would you use ?

Reply:
You can do:

```python
import nibabel as nib

def func(...) -> nib.Nifti1Image:
   pass
```